src/components/Pages/MainPage/MainPage.tsx

```diff
@@ -52,13 +65,57 @@ function MainPage(props: Props) {
       case AccountTabs.ASSETS:
         navigate(locations.account(profileAddress ?? nullAddress, AccountTabs.ASSETS))
         return
+      case AccountTabs.REFERRAL:
+        navigate(locations.account(profileAddress ?? nullAddress, AccountTabs.REFERRAL))
+        return
       case AccountTabs.OVERVIEW:
         navigate(locations.account(profileAddress ?? nullAddress))
         return
     }
   }, [])
 
-  return isLoading ? (
+  useEffect(() => {
+    if (selectedTab === AccountTabs.REFERRAL && tabsRef.current) {
+      setTimeout(() => {
+        tabsRef.current?.scrollIntoView({ behavior: 'smooth', block: 'start' })
+      }, 100)
+    }
+  }, [selectedTab])
+
+  if (selectedTab === AccountTabs.REFERRAL && isLoggedIn) {
```

Comment:
This `MainPage` renders the site, which all its tabs and such, and then renders each page according to the selected tab in the condition down below. Would you mind explaining to me why are we early returning here?